tasks/test_e2e.yml

```diff
@@ -0,0 +1,61 @@
+- name: copy shell script that executes prometheus query for collectd metrics
+  copy:
+    src: "{{ item }}"
+    dest: "/tmp"
+    owner: root
+    group: root
+    mode: 0755
+  with_items:
+    ['collectd_prometheus_query.sh']
+
+- name: execute promethues collectd query 
+  shell: "/bin/bash /tmp/collectd_prometheus_query.sh &"
+  register: checkmyconf
+
+- name: "[Debug] Execute Prometheus collectd query"
+  debug:
+    var: checkmyconf
+
+- name: "[Test] Check whether collectd cpu metrics received in prometheus for all overcloud nodes"
+  command: egrep 'controller-0|controller-1|controller-2|compute-0|compute-1|ceph-0' /tmp/query_collectd_cpu_percent
+  register: checkmyconf
+  check_mode: no
+  changed_when: no
+
+- name: "[Debug] Check whether collectd cpu metrics log created and received in Prometheus."
+  debug:
+    var: checkmyconf
+
+- name: "[Test] Check whether collectd cpu metrics log created and received in Prometheus"
+  debug: msg="COLLECTD CPU METRICS RECEIVED FOR ALL OVERCLOUD NODES"
+  when: checkmyconf.rc == 0
+
+
+- name: "[Test] Check whether collectd cpu ceph metrics received in prometheus for all ceph nodes"
+  command: egrep ceph-0 /tmp/cpu_ceph_bytes
+  register: checkmyconf
+  check_mode: no
+  changed_when: no
+
+- name: "[Debug] Check whether collectd cpu ceph metrics received in prometheus for all ceph nodes"
+  debug:
+    var: checkmyconf
+
+
+- name: "[Test] Check whether collectd cpu ceph metrics log created and received  in Prometheus"
+  debug: msg="COLLECTD CPU CEPH METRICS RECEIVED FOR ALL CEPH NODES"
+  when: checkmyconf.rc == 0
```

Comment:
You can replace these three tasks with a single task that used the failed_when option for the task.
```
- name: "[Test] Check whether collectd cpu ceph metrics received in prometheus for all ceph nodes"
  command: egrep ceph-0 /tmp/cpu_ceph_bytes
  register: checkmyconf
  check_mode: no
  failed_when:
    - check_my_conf.rc != 0
```

IMHO, this update is easier to grok, and makes it easier to see the different tests, and does what you want, as far as I can tell.